internal/service/profile/roles.go

```diff
@@ -0,0 +1,34 @@
+package profile
+
+import (
+	"github.com/rs/zerolog/log"
+	"github.com/ucl-arc-tre/portal/internal/rbac"
+	"github.com/ucl-arc-tre/portal/internal/types"
+)
+
+func (s *Service) updateApprovedResearcherStatus(user types.User) error {
```

Comment:
Would it make sense to shift some of this onus to the front end? So if both the training cert and agreement have an `agreed===true` then we call the function to update?

Reply:
ah gotcha. it's just what's there to stop someone doing a manual request to `POST /profile/roles` and assigning themselves as an approved researcher without actually completing the agreement or training?

> Will it matter if they do both in quick succession?

I _think_ there's a potential race condition, but doubt It'd be hit. I think you'd need to submit the training and <few hundred ms later agree to the training

Reply:
[I couldn't resist looking](https://gorm.io/docs/advanced_query.html#Locking)